src/lib/containers/widgets/facet-nav/SelectionCriteriaPill.tsx

```diff
@@ -1,83 +1,96 @@
-import * as React from 'react'
+import { Close } from '@material-ui/icons'
+import React from 'react'
 import { FunctionComponent } from 'react'
-
+import { SynapseConstants } from '../../../utils'
+import { unCamelCase } from '../../../utils/functions/unCamelCase'
 import {
   FacetColumnResult,
   FacetColumnResultRange,
   FacetColumnResultValueCount,
 } from '../../../utils/synapseTypes'
-import { unCamelCase } from '../../../utils/functions/unCamelCase'
 import { ElementWithTooltip } from '../ElementWithTooltip'
-import { SynapseConstants } from '../../../utils'
-import { Close } from '@material-ui/icons'
 
 export type FacetWithSelection = {
   facet: FacetColumnResult
   selectedValue?: FacetColumnResultValueCount
   displayValue?: string
 }
 
-export type SelectionCriteriaPillProps = {
-  facetWithSelection?: FacetWithSelection
-  className?: string
-  onRemove: Function
-  filter?: {
-    columnName: string
-    value: string
-  }
+export type SelectionCriteriaPillProps = (
+  | { readonly facetWithSelection: FacetWithSelection }
+  | {
+      filter: {
+        columnName?: string
+        value: string
+      }
+    }
+) & {
+  onRemove: () => void
 }
```

Comment:
Using a funky type to be a little more explicit about what we're expecting: `onRemove` with either `facetWithSelection` or `filter`, but not both.